metadata-ingestion/src/datahub/api/dataprocess/urn.py

```diff
@@ -0,0 +1,33 @@
+from dataclasses import dataclass
+
+import datahub.emitter.mce_builder as builder
+
+
+@dataclass(frozen=True, eq=True)
+class UrnBase:
+    id: str
+
+    @property
+    def urn(self) -> str:
+        raise NotImplementedError("Urn needs to be implemented")
+
+
+@dataclass(frozen=True, eq=True)
+class DataFlowUrn(UrnBase):
```

Comment:
aren't these added by https://github.com/datahub-project/datahub/pull/4618?

Reply:
I just merged this today afternoon and it was not here before. I think as a next step it would be to use this instead of string urns.